### ISSUE TITLE
Join group screen does not show child groups without filters

### ISSUE THREAD

**Issue by martinmeneval** [keycloak/keycloak#23582]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/ui

### Describe the bug

We've noticed this happening since our upgrade to Keycloak 22(.0.3), but it could be older.
When clicking on "Join Groups" in the Admin UI, either within the user creation page or on the Groups tab of a user's page, child groups don't appear. Using a filter does work to show child groups on nightly (we assume due to #23341 being fixed).

<img width="300" alt="Screenshot 2023-09-28 at 14 44 55" src="https://github.com/keycloak/keycloak/assets/34133448/ade3574c-5a73-4d9f-b8a9-09891d3f3def">
<img width="300" alt="Screenshot 2023-09-28 at 14 48 39" src="https://github.com/keycloak/keycloak/assets/34133448/d254f3f9-099a-40a0-b6e6-2ad975cbd2cf">  

Location of the two Join Groups buttons we're talking about.  

<img width="1019" alt="Screenshot 2023-09-28 at 16 10 22" src="https://github.com/keycloak/keycloak/assets/34133448/54b7c60b-663c-44d7-bea5-2f23ef2fc58d">  

Request showing that no subGroups are being returned when calling `/groups?first=0&max=11` 

### Version

22.0.3, but tested on last night's nightly

### Expected behavior

When clicking Join groups, we see the first 10 groups including subgroups:
 - A
   - Child of A-1
   - Child of A-2
   - Child of A-3
 - B
 - C
 - D
 - E
 - F
 - G

### Actual behavior

When clicking Join groups, we see the first 8 groups (not 10, even though the page size is 1 - 10), and no subgroups:
 - A
 - B
 - C
 - D
 - E
 - F
 - G
 - H

<img width="624" alt="Screenshot 2023-09-28 at 16 09 07" src="https://github.com/keycloak/keycloak/assets/34133448/ed3622bb-89ef-4099-a532-44fb3402d467">

Actual behaviour where fewer than 10 groups are returned, and no subgroups are.

<img width="579" alt="Screenshot 2023-09-28 at 16 14 10" src="https://github.com/keycloak/keycloak/assets/34133448/2650b438-3191-42ee-9277-7c9346ee4b85">

Even though there are more than 8 groups to list (without counting the subgroups)

### How to Reproduce?

1. Spin-up an instance of Keycloak nightly
2. Create a new realm
3. Create the following group structure:
    - A
      - Child of A-1
      - Child of A-2
      - Child of A-3
    - B
    - C 
    - D ... M
4. Create a user
5. Click Join groups on either the user creation page or the user's page - Groups tab.

### Anything else?

[This comment](https://github.com/keycloak/keycloak/issues/23341#issuecomment-1725605767) on #23341 may refer to the same problem.

**Comment by ssilvert** [comment]
I get odd behavior too, but I'm seeing something a little different.

I created groups 01, 02, 03, ... ,10, 11.  I put subgroups under 01.

I do see the first 10 groups in the dialog, 01 to 10.  But when I go to the next page, I see 01 and 11.  So there are two on the second page where I should have only seen group 11.

As for subgroups, I think the main problem is that there is no `>` indicator to show that subgroups exist.  This dialog is not meant to be a tree.  So searching and navigating to a subgroup with `>` is all that is required.

I also noticed that if you go to the Groups section, the tree view is not showing the `>` indicator unless you click on the group.
